src/Sylius/Bundle/UiBundle/Renderer/HtmlDebugTemplateBlockRenderer.php

```diff
@@ -23,30 +23,55 @@ public function __construct(private TemplateBlockRendererInterface $templateBloc
 
     public function render(TemplateBlock $templateBlock, array $context = []): string
     {
-        $shouldRenderHtmlDebug = strrpos($templateBlock->getTemplate(), '.html.twig') !== false;
+        $isTemplateBlockComponent = null !== $templateBlock->getComponent();
+        $shouldRenderHtmlDebug = $isTemplateBlockComponent || strrpos($templateBlock->getTemplate(), '.html.twig') !== false;
+
+        if (!$shouldRenderHtmlDebug) {
+            return $this->templateBlockRenderer->render($templateBlock, $context);
+        }
 
         $renderedParts = [];
 
-        if ($shouldRenderHtmlDebug) {
-            $renderedParts[] = sprintf(
-                '<!-- BEGIN BLOCK | event name: "%s", block name: "%s", template: "%s", priority: %d -->',
-                $templateBlock->getEventName(),
-                $templateBlock->getName(),
-                $templateBlock->getTemplate(),
-                $templateBlock->getPriority(),
-            );
+        if ($isTemplateBlockComponent) {
+            $renderedParts[] = $this->getBeginBlockForComponent($templateBlock);
+        } else {
+            $renderedParts[] = $this->getBeginBlockForTemplate($templateBlock);
         }
 
         $renderedParts[] = $this->templateBlockRenderer->render($templateBlock, $context);
-
-        if ($shouldRenderHtmlDebug) {
-            $renderedParts[] = sprintf(
-                '<!-- END BLOCK | event name: "%s", block name: "%s" -->',
-                $templateBlock->getEventName(),
-                $templateBlock->getName(),
-            );
-        }
+        $renderedParts[] = $this->getEndBlock($templateBlock);
 
         return implode("\n", $renderedParts);
     }
+
+    private function getBeginBlockForTemplate(TemplateBlock $templateBlock): string
```

Comment:
```suggestion
    private function getBeginBlock(TemplateBlock $templateBlock): string
```

Function should encapsulate this logic and not expose `$templateBlock` usage details into main loop at `render()` above.

Reply:
P.S. I see why it's done as `render()` currently makes a decision whenever `TemplateBlock` is a template or component, so copying this logic into `getBeginBlock` will be a duplication, but if a suggested `TemplateBlock` changes will be made, then it will be unnecessary.